include/lsst/afw/table/io/FitsSchemaInputMapper.h

```diff
@@ -0,0 +1,160 @@
+// -*- lsst-c++ -*-
+#ifndef AFW_TABLE_IO_FitsSchemaInputMapper_h_INCLUDED
+#define AFW_TABLE_IO_FitsSchemaInputMapper_h_INCLUDED
+
+#include "lsst/afw/fits.h"
+#include "lsst/afw/table/Schema.h"
+#include "lsst/afw/table/io/InputArchive.h"
+#include "lsst/afw/table/BaseRecord.h"
+
+namespace lsst { namespace afw { namespace table { namespace io {
+
+class FitsColumnReader {
+public:
+
+    FitsColumnReader() {}
+
+    // Neither copyable nor moveable.
+    FitsColumnReader(FitsColumnReader const &) = delete;
+    FitsColumnReader(FitsColumnReader &&) = delete;
+    FitsColumnReader & operator=(FitsColumnReader const &) = delete;
+    FitsColumnReader & operator=(FitsColumnReader &&) = delete;
+
+    virtual void readCell(
+        BaseRecord & record,
+        std::size_t row,
+        fits::Fits & fits,
+        PTR(InputArchive) const & archive
+    ) const = 0;
+
+    virtual ~FitsColumnReader() {}
+
+};
+
+/**
+ *  A structure that describes a field as a collection of related strings read from the FITS header.
+ */
+struct FitsSchemaItem {
+    int column;          // column number (0-indexed); -1 for Flag fields
+    int bit;             // flag bit number (0-indexed); -1 for non-Flag fields
+    std::string ttype;   // name of the field (from TTYPE keys)
+    std::string tform;   // FITS column format code (from TFORM keys)
+    std::string tccls;   // which field class to use (from our own TCCLS keys)
+    std::string tunit;   // field units (from TUNIT keys)
+    std::string doc;     // field docs (from comments on TTYPE keys)
+
+    explicit FitsSchemaItem(int column_, int bit_) : column(column_), bit(bit_) {}
+};
+
+
+/**
+ *  A class that describes a mapping from a FITS binary table and an afw::table Schema.
```

Comment:
"between a... and an... ", or "from a... to an...".

(I've not yet looked to see if the mapping is bidrectional.)